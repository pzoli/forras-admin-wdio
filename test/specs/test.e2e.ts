import { expect } from '@wdio/globals'
import LoginPage from '../pageobjects/login.page.js'
import SecurePage from '../pageobjects/secure.page.js'
import subjectsPage from '../pageobjects/subjects.page.js'

describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        await LoginPage.open()
        let username : string = process.env.username?process.env.username:''
        let passwd : string = process.env.passwd?process.env.passwd:''
        await LoginPage.login(username, passwd)
        await expect(SecurePage.kliensTable).toBeExisting()
        await expect(SecurePage.kliensTable).toHaveText(
            expect.stringContaining('kliens lista'))
        await expect(SecurePage.kliensTable).toMatchSnapshot('kliensTable')
    })
})
describe('Manage subject', () => {
    it('Add, filter, delete subject', async() => {
        await subjectsPage.open()
        await subjectsPage.addSubject("test-subj")
        await subjectsPage.filterSubject("test-subj")
        await browser.pause(500)

        await browser.saveScreenshot('./screenshot.png')

        await expect(subjectsPage.firstRowSubjectNameCol).toHaveText(
            expect.stringContaining('test-subj')
        )
        await expect(await subjectsPage.subjectTableRow(0)).toBeExisting()
        await subjectsPage.deleteSubject()
    })

    it('should logout', async () => {
        await LoginPage.logout()
    })
})

