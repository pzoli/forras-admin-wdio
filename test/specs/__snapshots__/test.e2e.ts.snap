// Snapshot v1

exports[`My Login application > should login with valid credentials > flashAlert 1`] = `"<div class="ui-datatable-header ui-widget-header ui-corner-top">kliens lista</div>"`;
